packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -158,6 +158,48 @@ const isWithinAvailableHours = (
   return false;
 };
 
+/**
+ * Function to check if a certain timeslot is between the date overrides an user has.
+ *
+ * @param timeSlot
+ * @param dateOverrides
+ */
+const isWithinDateOverrides = (
+  timeSlot: { start: ConfigType; end: ConfigType },
+  {
+    dateOverrides,
+  }: {
+    dateOverrides: {
+      start: Date;
+      end: Date;
+    }[];
+  }
+) => {
+  const timeSlotStart = dayjs(timeSlot.start).utc();
+  const timeSlotEnd = dayjs(timeSlot.end).utc();
+
+  for (const dateOverride of dateOverrides) {
+    const startTime = timeSlotStart
+      .startOf("day")
+      .add(dateOverride.start.getUTCHours(), "hour")
+      .add(dateOverride.start.getUTCMinutes(), "minute");
```

Comment:
Will this be affected by the DST bug? cc @roae to investigate 🙏🏽

Reply:
Yes, here we must always keep in mind the time zone of the event/organizer and the booker.

We need to merge main in this branch to bring those fixes and apply them in this function. I will do it